src/modules/connections/connection-manager.service.ts

```diff
@@ -0,0 +1,46 @@
+import { Injectable } from '@nestjs/common';
+import {ConnectionsRepositoryService} from './connections-repository.service';
+import { Connection, ConnectionOptions, createConnection } from 'typeorm';
+import {ConnectionData} from './connections.interfaces';
+
+@Injectable()
+export class ConnectionManagerService {
+  private connections: Map<number, Connection> = new Map();
+  constructor(
+    private connectionRepository: ConnectionsRepositoryService,
+  ) {}
+
+  async getConnection(id: number): Promise<Connection> {
+    let connection = this.connections.get(id);
+    try {
+      if (connection && connection.isConnected) {
+        return connection;
+      } else if (connection) {
+        return await connection.connect();
+      }
+      const connectionData: ConnectionData = await this.connectionRepository.getDataForConnectionCreating(id);
+      if (!connectionData) {
+        return null;
+      }
+      connection = await createConnection(connectionData as ConnectionOptions);
+      this.connections.set(id, connection);
+      return connection;
+    } catch (e) {
+      return null;
```

Comment:
Same here
And add console.log everywhere in catch, so we could find out an error in case of some trouble )

Reply:
done